src/utils.js

```diff
@@ -136,7 +136,30 @@ export const isPruneTraceOff = () =>
   );
 
 export const isSwitchedOff = () => {
-  return TracerGlobals.getTracerInputs().switchOff;
+  return TracerGlobals.getTracerInputs().switchOff || !isValidAlias();
+};
+
+export const getValidAliasesOrEmptyArray = () =>
+  process.env['VALID_ALIASES'] || [];
+
+export const getHandlerContext = () =>
+  TracerGlobals.getHandlerInputs().context || {};
+
+export const getInvokedArn = () => getHandlerContext().invokedFunctionArn || '';
+export const getInvokedVersion = () =>
+  getHandlerContext().functionVersion || '';
+
+export const getInvokedAliasOrNull = () =>
+  safeExecute(() => {
+    return getInvokedArn().split(':').length >= 8
+      ? getInvokedArn().split(':')[7]
+      : null;
+  })();
+
+export const isValidAlias = () => {
+  const validAliases = getValidAliasesOrEmptyArray();
+  const currentAlias = getInvokedAliasOrNull();
+  return validAliases.length === 0 || validAliases.includes(currentAlias);
```

Comment:
The aliases here are `contain`, so if the user has two aliases say `prod` and `noProd` if they will use `prod` they will catch both aliases, right?
I don't mind having it, but please make sure the customer does not have aliases with the structure I mentioned

Reply:
I don't understand

Reply:
It's related to `getValidAliasesOrEmptyArray` and on the return value, if you are returning string then includes is like contains `XYZ.contains(XY) is true`, but `['XYZ'].contains('XY') is false`

Reply:
fixed that, only arrays now